lib/Map/TableStructure.js

```diff
@@ -503,6 +514,118 @@ TableStructure.prototype.getIdMapping = function() {
     }, {});
 };
 
+/**
+ * Appends table2 to this table. If rowNumbers are provided, only takes those
+ * row numbers from table2.
+ * Changes all the columns in one go, to avoid partial updates from tracked values.
+ * @param {TableStructure} table2 The table to add to this one.
+ * @param {Integer[]} [rowNumbers] The row numbers from table2 to add (defaults to all).
+ */
+TableStructure.prototype.append = function(table2, rowNumbers) {
+    if (this.columns.length !== table2.columns.length) {
+        throw new DeveloperError('Cannot add tables with different numbers of columns.');
+    }
+    var updatedColumnValueses = [];
```

Comment:
`updatedColumnValuesArrays` maybe? When I read 'valueses' I tend to apply gollum-logic and assume it's the same as 'values' ;)